azure-mgmt-monitor/src/test/java/com/microsoft/azure/management/redis/MonitorActivityAndMetricsTests.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.redis;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.management.compute.VirtualMachine;
+import com.microsoft.azure.management.monitor.EventData;
+import com.microsoft.azure.management.monitor.MetricCollection;
+import com.microsoft.azure.management.monitor.MetricDefinition;
+import com.microsoft.azure.management.monitor.ResultType;
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+
+public class MonitorActivityAndMetricsTests extends MonitorManagementTest {
+    @Test
+    public void canListEventsAndMetrics() throws Exception {
+        DateTime recordDateTime = DateTime.parse("2018-01-24T00:07:40.350Z");
+        VirtualMachine vm = computeManager.virtualMachines().list().get(0);
+
+        // Metric Definition
+        List<MetricDefinition> mt = monitorManager.metricDefinitions().listByResource(vm.id());
+
+        Assert.assertNotNull(mt);
+
+        // Metric
+        MetricCollection metrics = mt.get(0).defineQuery()
+                .withStartTime(recordDateTime.minusDays(30))
+                .withEndTime(recordDateTime)
+                .withResultType(ResultType.DATA)
+                .execute();
+
+        Assert.assertNotNull(metrics);
+
+        // Activity Logs
+        PagedList<EventData> retVal = monitorManager.activityLogs()
+                .defineQuery()
+                .withStartTime(recordDateTime.minusDays(30))
+                .withEndTime(recordDateTime)
+                .defineResponseProperties()
+                    .withResourceId()
+                    .withEventTimestamp()
+                    .withOperationName()
+                    .withEventName()
+                    .apply()
```

Comment:
This is wrong; you want to "attach()" your response properties to your query than execute it. Again, thing of SELECT, you define all the things you want to be returned, filtered by etc, than you have a separate step to "execute"/"run" your full query.

Plus "apply" has a whole different meaning in Fluent; it's used as part of the "update" pattern and only for the root resource. Attaching new children to an update of a parent resource will use the ".parent()" pattern and not ".attach()" or ".apply()".